### ISSUE TITLE
Duplicate records in .fides/db_dataset.yml

### ISSUE THREAD

**Issue by ewdurbin** [ethyca/fides#3351]
### Bug Description

While writing tests for https://github.com/ewdurbin/dbml-to-fides, I grabbed `.fides/db_dataset.yml` as a fixture.

When testing some transformations, I discovered that this file contains duplicates for `fidesuser`, `fidesuserpermissions`, and `client` tables with conflicting information (namely `data_categories`).

#### Steps to Reproduce

```shell
$ egrep '  - name: fidesuser$|  - name: fidesuserpermissions$|  - name: client$' .fides/db_dataset.yml 
  - name: client
  - name: fidesuser
  - name: fidesuserpermissions
  - name: client
  - name: fidesuser
  - name: fidesuserpermissions
```

### Expected behavior

I would have expected that something validates a Fides dataset to have non-conflicting keys for each member of a `collection`

### Screenshots

n/a

### Environment

n/a

### Additional context

n/a

**Comment by ewdurbin** [comment]
I would definitely prefer to see validation in `fideslang`, as it's a lighter weight dependency for utilities like my use case in `dbml-to-fides`.

**Comment by ThomasLaPiana** [comment]
Agreed, can add a helper function like `verify_unique_keys` and reuse it is a validator. I'm going to move this issue over to `fideslang`
